quil-rs/src/parser/command.rs

```diff
@@ -189,16 +190,17 @@ pub(crate) fn parse_defcal_gate<'a>(
     let (input, qubits) = many0(parse_qubit)(input)?;
     let (input, _) = token!(Colon)(input)?;
     let (input, instructions) = parse_block(input)?;
-    Ok((
-        input,
-        Instruction::CalibrationDefinition(Calibration {
-            instructions,
-            modifiers,
-            name,
-            parameters,
-            qubits,
-        }),
-    ))
+    let calibration = Calibration::new(name.as_str(), parameters, qubits, instructions, modifiers)
```

Comment:
Within the parser, there should remain a fast path which skips redundant validation. That is, identifiers have already been parsed according to the spec; they don't need to be re-validated with regex. This isn't a nit; parsing is on the critical path for translating programs and thus program execution, and should remain fast. Submitted programs are _often_ on the order of 10k lines long.

Reply:
I will avoid using these validating constructors in the parser then. I thought it would help ensure some consistency between parsing and programmatic construction of a Program, but I see your point about it being on the critical path and not wanting to waste cycles doing redundant checks.